mail/dovecot/Portfile

```diff
@@ -1,110 +1,453 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
 PortSystem          1.0
+PortGroup           github 1.0
 
+github.setup        dovecot core 2.3.9.2
 name                dovecot
+revision            0
 conflicts           dovecot2
-epoch               20060722
-version             1.2.17
-revision            2
-set branch          [join [lrange [split ${version} .] 0 1] .]
+epoch               20200128
```

Comment:
There is no need to change the `epoch`.

Reply:
I’m confused. I thought that the purpose of `epoch` was to show that an upgrade is needed. If it doesn’t change, then doesn’t that defeat the purpose?

I’ll change it back, but confusedly.

Reply:
It is only needed if the versioning scheme changes in a way that the `version` number alone would not indicate an update. In this case, the fact that the prior  `dovecot` port had an epoch means all future versions need an `epoch` that is the same or higher (because a null `epoch` is effectively `epoch=0`) but since in this case the core version is going from 1.x to 2.x, the `epoch` need not be changed.

Reply:
`epoch` is only changed if necessary, for example to enforce a `version` downgrade when our version compare algorithm made a mistake (corner cases) or the newer version is buggy. A newer `version` will let users upgrade as usual.

From the [guide](https://guide.macports.org/):
> **epoch**
> An optional keyword (default value is 0) that must be used when a port is updated to a version that is numerically less than the previous version, for example 1.10 -> 1.2 or 20070928 -> 1.0. [...]